homeassistant/components/mobile_app/__init__.py

```diff
@@ -52,41 +56,20 @@
         except ValueError:
             pass
 
-    hass.async_create_task(discovery.async_load_platform(
+    hass.async_create_task(async_load_platform(
         hass, 'notify', DOMAIN, {}, config))
 
     return True
 
 
 async def async_setup_entry(hass, entry):
     """Set up a mobile_app entry."""
-    registration = entry.data
+    app_id = entry.data[ATTR_APP_ID]
 
-    webhook_id = registration[CONF_WEBHOOK_ID]
+    hass.data[DOMAIN][DATA_CONFIG_ENTRIES][app_id] = entry
 
-    hass.data[DOMAIN][DATA_CONFIG_ENTRIES][webhook_id] = entry
-
-    device_registry = await dr.async_get_registry(hass)
-
-    identifiers = {
-        (ATTR_DEVICE_ID, registration[ATTR_DEVICE_ID]),
-        (CONF_WEBHOOK_ID, registration[CONF_WEBHOOK_ID])
-    }
-
-    device = device_registry.async_get_or_create(
-        config_entry_id=entry.entry_id,
-        identifiers=identifiers,
-        manufacturer=registration[ATTR_MANUFACTURER],
-        model=registration[ATTR_MODEL],
-        name=registration[ATTR_DEVICE_NAME],
-        sw_version=registration[ATTR_OS_VERSION]
-    )
-
-    hass.data[DOMAIN][DATA_DEVICES][webhook_id] = device
-
-    registration_name = 'Mobile App: {}'.format(registration[ATTR_DEVICE_NAME])
-    webhook_register(hass, DOMAIN, registration_name, webhook_id,
-                     handle_webhook)
+    for registration in hass.data[DOMAIN][DATA_REGISTRATIONS].values():
```

Comment:
If each app gets it's own entry, then each entry will trigger here each registration to be set up again, and triggering the save.